pkg/cmd/experimental/diagnostics/diagnostics.go

```diff
@@ -116,45 +118,48 @@ func NewCommandDiagnostics(name string, fullName string, out io.Writer) *cobra.C
 	cmd.Flags().BoolVar(&o.IsHost, options.FlagIsHostName, false, "look for systemd and journald units even without master/node config")
 	flagtypes.GLog(cmd.Flags())
 	options.BindLoggerOptionFlags(cmd.Flags(), o.LogOptions, options.RecommendedLoggerOptionFlags())
-	options.BindDiagnosticFlag(cmd.Flags(), &o.RequestedDiagnostics, options.NewRecommendedDiagnosticFlag())
 
 	return cmd
 }
 
 // Complete fills in DiagnosticsOptions needed if the command is actually invoked.
-func (o *DiagnosticsOptions) Complete() error {
+func (o *DiagnosticsOptions) Complete(args []string) error {
 	var err error
 	o.Logger, err = o.LogOptions.NewLogger()
 	if err != nil {
 		return err
 	}
 
-	return nil
-}
+	// If not given master/client config file locations, check if the defaults exist
+	// and adjust the options accordingly:
+	if len(o.MasterConfigLocation) == 0 {
+		if _, err := os.Stat(StandardMasterConfigPath); !os.IsNotExist(err) {
+			o.MasterConfigLocation = StandardMasterConfigPath
```

Comment:
@stevekuznetsov the master file checked is noted in the diagnostic output if anything goes wrong, e.g.:

```
WARN:  [DH0005 from diagnostic MasterConfigCheck@openshift/origin/pkg/diagnostics/host/check_master_config.go:58]
       Validation of master config file '/openshift.local.config/master/master-config.yaml' warned:
       assetConfig.loggingPublicURL: invalid value '', Details: required to view aggregated container logs in the console
       assetConfig.metricsPublicURL: invalid value '', Details: required to view cluster metrics in the console
```